state-chain/pallets/cf-funding/src/tests.rs

```diff
@@ -596,14 +594,13 @@ fn redemption_expiry_removes_redemption() {
 		);
 
 		assert_ok!(Funding::redemption_expired(RuntimeOrigin::root(), ALICE, Default::default()));
-		assert!(PendingRedemptionsRestrictedBalance::<Test>::get(&ALICE).is_none());
 
 		// Tax was paid, rest is returned.
 		assert_eq!(Flip::total_balance_of(&ALICE), TOTAL_FUNDS - REDEMPTION_TAX);
-		// Restricted funds are restricted again, minus redemption tax.
+		// Restricted funds are restricted again.
 		assert_eq!(
 			*RestrictedBalances::<Test>::get(&ALICE).get(&RESTRICTED_ADDRESS).unwrap(),
-			RESTRICTED_AMOUNT - REDEMPTION_TAX
+			RESTRICTED_AMOUNT
```

Comment:
Actually, strictly speaking this was correct - the redemption tax *is* paid by the restricted funds. If we want to be totally correct, we should include this in our calcuation, ie line 464 of lib.rs:

```
restricted: min(total_restricted_balance, debit_amount)
```

should be something like:

```
restricted: min(total_restricted_balance, debit_amount).saturating_sub(redemption_tax)
```

Reply:
Why is this the case? The redemption tax should not affect the restricted amount in case of an expiry. In any case, the amount has to reset to the amount before the redemption attempt. Everything else seems a bit odd to me.